src/foam/java/refinements.js

```diff
@@ -436,6 +436,10 @@ foam.CLASS({
       m.body = 'return ( o instanceof Number ) ?'
             + '((Number)o).intValue() :'
             + '(int)o;';
+
+      var c = info.getMethod('comparePropertyToObject');
+      c.body = 'return compareValues((int) key, (int) get_(o));';
```

Comment:
The second cast isn't required because that's already the declared return value of get_(). Here and in other implementations also.